src/__tests__/cat.test.ts

```diff
@@ -0,0 +1,233 @@
+      expect(cat2.resps).toEqual([0, 1, 0, 1, 1, 1, 1]);
+    });
+
+    it('correctly updates zetas', () => {
+      expect(cat2.zetas).toEqual([
+        convertZeta({ a: 1, b: -0.447, c: 0.5, d: 1 }, format),
+        convertZeta({ a: 1, b: 2.869, c: 0.5, d: 1 }, format),
+        convertZeta({ a: 1, b: -0.469, c: 0.5, d: 1 }, format),
+        convertZeta({ a: 1, b: -0.576, c: 0.5, d: 1 }, format),
+        convertZeta({ a: 1, b: -1.43, c: 0.5, d: 1 }, format),
+        convertZeta({ a: 1, b: -1.607, c: 0.5, d: 1 }, format),
+        convertZeta({ a: 1, b: 0.529, c: 0.5, d: 1 }, format),
+      ]);
+    });
+
+    it.each`
+      deepCopy
+      ${true}
+      ${false}
+    `("correctly suggests the next item (closest method) with deepCopy='$deepCopy'", ({ deepCopy }) => {
+      const expected = { nextStimulus: s5, remainingStimuli: [s4, s1, s3, s2] };
+      const received = cat1.findNextItem(stimuli, 'closest', deepCopy);
+      expect(received).toEqual(expected);
+    });
+
+    it.each`
+      deepCopy
+      ${true}
+      ${false}
+    `("correctly suggests the next item (mfi method) with deepCopy='$deepCopy'", ({ deepCopy }) => {
+      const expected = { nextStimulus: s1, remainingStimuli: [s4, s5, s3, s2] };
+      const received = cat3.findNextItem(stimuli, 'MFI', deepCopy);
+      expect(received).toEqual(expected);
+    });
+
+    it.each`
+      deepCopy
+      ${true}
+      ${false}
+    `("correctly suggests the next item (middle method) with deepCopy='$deepCopy'", ({ deepCopy }) => {
+      const expected = { nextStimulus: s1, remainingStimuli: [s4, s5, s3, s2] };
+      const received = cat5.findNextItem(stimuli, undefined, deepCopy);
+      expect(received).toEqual(expected);
+    });
+
+    it.each`
+      deepCopy
+      ${true}
+      ${false}
+    `("correctly suggests the next item (fixed method) with deepCopy='$deepCopy'", ({ deepCopy }) => {
+      expect(cat8.itemSelect).toBe('fixed');
+      const expected = { nextStimulus: s1, remainingStimuli: [s2, s3, s4, s5] };
+      const received = cat8.findNextItem(stimuli, undefined, deepCopy);
+      expect(received).toEqual(expected);
+    });
+
+    it.each`
+      deepCopy
+      ${true}
+      ${false}
+    `("correctly suggests the next item (random method) with deepCopy='$deepCopy'", ({ deepCopy }) => {
+      let received;
+      const stimuliSorted = stimuli.sort((a: Stimulus, b: Stimulus) => a.difficulty! - b.difficulty!); // ask
+      let index = Math.floor(rng() * stimuliSorted.length);
+      received = cat4.findNextItem(stimuliSorted, undefined, deepCopy);
+      expect(received.nextStimulus).toEqual(stimuliSorted[index]);
+
+      for (let i = 0; i < 3; i++) {
+        const remainingStimuli = received.remainingStimuli;
+        index = Math.floor(rng() * remainingStimuli.length);
+        received = cat4.findNextItem(remainingStimuli, undefined, deepCopy);
+        expect(received.nextStimulus).toEqual(remainingStimuli[index]);
+      }
+    });
```

Comment:
`deepCopy` is part of the original jsCat code. It dictates whether the input array should be deep copied before sorting in the `findNextItem` method. While the `deepCopy` parameter itself isn't new (it's been around since [this commit](https://github.com/yeatmanlab/jsCAT/commit/8c7611325f52c4151943327d480b921f89643538#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80), Emily noticed that it wasn't being tested, so now the unit tests test whether `findNextItem` works as expected with both `deepCopy=true` and `deepCopy=false`.